File: common/common-core/src/main/java/com/glowxq/core/common/enums/MenuType.java
package com.glowxq.core.common.enums;

import com.glowxq.core.common.enums.base.BaseType;
import com.glowxq.core.common.exception.common.AlertsException;
import lombok.AllArgsConstructor;
import lombok.Getter;

/**
 * @author glowxq
 * @version 1.0
 * @date 2024/6/7
 */
@Getter
@AllArgsConstructor
public enum MenuType implements BaseType {

    Client("Client", "客户端"),
    Admin("Admin", "管理端"),
    Common("Common", "通用");

    /**
     * CODE
     */
    private final String code;

    /**
     * 名字
     */
    private final String name;

    /**
     * 匹配代码
     *
     * @param code
     * @return {@link MenuType}
     */
    public static MenuType matchCode(String code) {
        for (MenuType type : values()) {
            if (type.getCode().equals(code)) {
                return type;
            }
        }
        throw new AlertsException(String.format("%s 未匹配到对应 DataScopeType 枚举类型", code));
    }
}

File: oj-web/src/enums/oj/system/MenuType.ts
import type { BaseEnum } from '@/enums/base';

// 菜单类型枚举接口
interface MenuTypeEnum extends BaseEnum {
  tooltip?: string;
  text?: string;
}

/**
 * 菜单类型枚举
 * 用于标识不同的菜单类型
 */
export const MenuType: { [key: string]: MenuTypeEnum } = {
  /**
   * 客户端菜单
   */
  CLIENT: {
    code: 'Client',
    name: '客户端',
    tooltip: '客户端菜单',
    text: '在客户端前台显示的菜单'
  },

  /**
   * 管理后台菜单
   */
  ADMIN: {
    code: 'Admin',
    name: '管理端',
    tooltip: '管理端菜单',
    text: '在管理后台显示的菜单'
  },

  /**
   * 通用菜单
   */
  COMMON: {
    code: 'Common',
    name: '通用',
    tooltip: '通用菜单',
    text: '在客户端和管理端都显示的菜单'
  }
};

/**
 * 根据code获取枚举
 * @param code 菜单类型代码
 * @returns 枚举值或null
 */
export function matchCode(code: string): MenuTypeEnum | null {
  return Object.values(MenuType).find(item => item.code === code) || null;
}

